pkgs/data/themes/graphite/default.nix

```diff
@@ -0,0 +1,91 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, gnome-themes-extra
+, gtk-engine-murrine
+, jdupes
+, sassc
+, themeVariants ? [] # default: blue
+, colorVariants ? [] # default: all
+, sizeVariants ? [] # default: standard
+, tweaks ? []
+, wallpapers ? false
+}:
+
+let
+  assertSubListOf = name: val: xs:
+    let
+      unexpected = lib.subtractLists xs val;
+    in
+      lib.assertMsg (unexpected == [])
+        "${name} must be a sublist of ${
+          lib.generators.toPretty {} xs}, but contains: ${
+            lib.generators.toPretty {} unexpected}";
+
+in
+assert assertSubListOf "theme variants" themeVariants [ "default" "purple" "pink" "red" "orange" "yellow" "green" "teal" "blue" "all" ];
+assert assertSubListOf "color variants" colorVariants [ "standard" "light" "dark" ];
+assert assertSubListOf "size variants" sizeVariants [ "standard" "compact" ];
+assert assertSubListOf "tweaks" tweaks [ "nord" "black" "midblack" "rimless" "normal" ];
```

Comment:
Based on the discussion in https://github.com/NixOS/nixpkgs/pull/153740, it may be better to use throw instead of assertions here. See https://github.com/NixOS/nixpkgs/pull/153740#issuecomment-1007260798.

Reply:
Done.